File: fix_const.py
import re

file_path = r"e:\d\DOCVOICE-ORG\docvoice\lib\web_extension\screens\extension_settings_screen.dart"

with open(file_path, "r", encoding="utf-8") as f:
    content = f.read()

# 1. Fix undefined 'theme' variable
content = content.replace("theme.scaffoldBackgroundColor", "Theme.of(context).scaffoldBackgroundColor")
content = content.replace("theme.cardTheme.color", "Theme.of(context).cardTheme.color")

# 2. Add 'final theme = Theme.of(context);' if I missed it? We replaced it with Theme.of(context) directly, so that's fine.

# 3. Remove 'const' before Text, TextStyle, Divider, Icon if they contain Theme.of(context)
content = re.sub(r'const\s+TextStyle\(([^)]*?Theme\.of\(context\)[^)]*?)\)', r'TextStyle(\1)', content)
# Handle multiline Text and TextStyle consts
content = re.sub(r'const\s+Text\(([\s\S]*?Theme\.of\(context\)[\s\S]*?)\)', r'Text(\1)', content)
content = re.sub(r'const\s+TextStyle\(([\s\S]*?Theme\.of\(context\)[\s\S]*?)\)', r'TextStyle(\1)', content)
content = re.sub(r'const\s+Divider\(([\s\S]*?Theme\.of\(context\)[\s\S]*?)\)', r'Divider(\1)', content)
content = re.sub(r'const\s+Icon\(([\s\S]*?Theme\.of\(context\)[\s\S]*?)\)', r'Icon(\1)', content)

# Some specific replacements for constant collections or deeper trees
# Fallback replacement for any const TextStyle or Text that still remains with Theme.of
lines = content.split('\n')
for i in range(len(lines)):
    if 'Theme.of(context)' in lines[i]:
        # remove const from the same line if it causes issues
        lines[i] = re.sub(r'const\s+Text\(', 'Text(', lines[i])
        lines[i] = re.sub(r'const\s+TextStyle\(', 'TextStyle(', lines[i])
        lines[i] = re.sub(r'const\s+Icon\(', 'Icon(', lines[i])
        lines[i] = re.sub(r'const\s+Divider\(', 'Divider(', lines[i])

# Join back
content = '\n'.join(lines)

# If 'const' is on a previous line, the multiline regex should have caught it, but let's be safe.
# Another pass for multiline const Text/TextStyle
content = re.sub(r'const\s+Text\s*\(\s*(?:"[^"]*"|''[^'']*'')\s*,\s*style:\s*(?:const\s+)?TextStyle\s*\([\s\S]*?Theme\.of\(context\)[\s\S]*?\)\s*\)', lambda m: m.group(0).replace('const Text', 'Text').replace('const TextStyle', 'TextStyle'), content)

# Replace specific known errors
content = content.replace('style: const TextStyle(color:', 'style: TextStyle(color:')
content = content.replace('style: const\n                                      TextStyle', 'style: TextStyle')
content = content.replace('style: const\n                      TextStyle', 'style: TextStyle')
content = content.replace('style: const TextStyle(fontSize:', 'style: TextStyle(fontSize:')

with open(file_path, "w", encoding="utf-8") as f:
    f.write(content)

File: fix_const.dart
import 'dart:io';

void main() {
  final file = File('lib/web_extension/screens/extension_settings_screen.dart');
  if (!file.existsSync()) {
    print('File not found');
    return;
  }
  
  String content = file.readAsStringSync();

  // Fix theme variable
  content = content.replaceAll('theme.scaffoldBackgroundColor', 'Theme.of(context).scaffoldBackgroundColor');
  content = content.replaceAll('color: theme.cardTheme.color ?? const Color(0xFF1E1E1E),', 'backgroundColor: Theme.of(context).cardTheme.color ?? const Color(0xFF1E1E1E),');
  content = content.replaceAll('color: theme.cardTheme.color,', 'color: Theme.of(context).cardTheme.color,');

  // Fix AlertDialog context issue (ctx vs context) in _resetMacrosToDefaults
  content = content.replaceAll('color: Theme.of(ctx).cardTheme.color', 'backgroundColor: Theme.of(ctx).cardTheme.color');
  // Wait, my replaceAll above would have done Theme.of(context), let's fix it for ctx
  content = content.replaceAll('backgroundColor: Theme.of(context).cardTheme.color ?? const Color(0xFF1E1E1E),', 'backgroundColor: Theme.of(context).cardTheme.color ?? const Color(0xFF1E1E1E),');
  // Actually, we can just replace 'color: theme.cardTheme.color' to 'backgroundColor: Theme.of(context).cardTheme.color' where 'color: theme.cardTheme.color' was used for AlertDialog
  content = content.replaceAll('Theme.of(context).cardTheme.color ??', 'Theme.of(ctx).cardTheme.color ??');
  content = content.replaceAll('Theme.of(ctx).cardTheme.color ??', 'Theme.of(context).cardTheme.color ??'); // Put back for the first one. Let's do it right.

  List<String> lines = content.split('\n');
  
  for (int i = 0; i < lines.length; i++) {
    if (lines[i].contains('theme.cardTheme.color')) {
      // should be replaced already, but just in case
    }
    
    // In _resetMacrosToDefaults, context is ctx
    if (lines[i].contains('builder: (ctx) => AlertDialog(')) {
       if (lines[i+1].contains('Theme.of(context)')) {
         lines[i+1] = lines[i+1].replaceAll('Theme.of(context)', 'Theme.of(ctx)');
       }
    }

    // Remove const if Theme.of(context) is on the same line
    if (lines[i].contains('Theme.of(context)') || lines[i].contains('Theme.of(ctx)')) {
      lines[i] = lines[i].replaceAll('const Text(', 'Text(');
      lines[i] = lines[i].replaceAll('const TextStyle(', 'TextStyle(');
      lines[i] = lines[i].replaceAll('const Icon(', 'Icon(');
      lines[i] = lines[i].replaceAll('const Divider(', 'Divider(');
    }
  }
  
  content = lines.join('\n');
  file.writeAsStringSync(content);
  print('Fixed constants successfully');
}
